packages/typespec-vscode/src/code-action-provider.ts

```diff
@@ -47,9 +49,97 @@ export class TypeSpecCodeActionProvider implements vscode.CodeActionProvider {
         );
       }
     });
+
+    // When the corresponding node dependency package is not installed,
+    // consider generating a quick fix to install via npm command
+    const diagnostics = context.diagnostics.filter(
+      (diagnostic) =>
+        diagnostic.source === "TypeSpec" &&
+        diagnostic.code &&
+        diagnostic.code === "import-not-found",
+    );
+    if (diagnostics.length === 0) {
+      return actions;
+    }
+
+    const uris = await vscode.workspace
```

Comment:
you should be able to find the package.json from the diagnstoic. The package.json from the whole project may not be related to the diagnstoic.